File: src/backend/location-service/src/main/java/com/swiftMove/locationservice/model/Address.java
package com.swiftMove.locationservice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "addresses")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(name = "line1")
    private String line1;

    @Column(name = "line2", nullable = true)
    private String line2;

    @Column(name = "city")
    private String city;

    @Column(name = "state_or_province")
    private String stateOrProvince;

    @Column(name = "country")
    private String country;

    @Column(name = "postal_or_zip_code")
    private String postalOrZipCode;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Instant updatedAt;
}

File: src/swiftmove-dashboards/src/types/old-frontend/address.ts
export interface Address {
  id: string | number;
  line1: string;
  line2?: string;
  city: string;
  stateOrProvince: string;
  country: string;
  postalOrZipCode: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface AddressFormData {
  line1: string;
  line2?: string;
  city: string;
  stateOrProvince: string;
  country: string;
  postalOrZipCode: string;
}
